File: android/src/main/java/com/blitzapp/animatedsplash/animation/ObjectAnimation.java
package com.blitzapp.animatedsplash.animation;


import android.animation.Animator;
import android.animation.AnimatorListenerAdapter;
import android.animation.ObjectAnimator;
import android.animation.PropertyValuesHolder;
import android.animation.ValueAnimator;
import android.util.Log;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.LinearInterpolator;
import android.view.animation.TranslateAnimation;

import static android.content.ContentValues.TAG;
import static com.blitzapp.animatedsplash.AnimationsList.animationsList;
import static com.blitzapp.animatedsplash.HideAnimationList.hideAnimationListHead;
import static com.blitzapp.animatedsplash.HideAnimationList.hideAnimationsList;
import static com.blitzapp.animatedsplash.animation.Splash.allExecuted;
import static com.blitzapp.animatedsplash.animation.Splash.allHideExecuted;
import static com.blitzapp.animatedsplash.animation.Splash.animationhide;
import static com.blitzapp.animatedsplash.animation.Splash.hide;
import static com.blitzapp.animatedsplash.animation.Splash.runAnimation;
import static com.blitzapp.animatedsplash.animation.Splash.runHideAnimation;
import static com.blitzapp.animatedsplash.animation.Splash.runSpecificAnimation;
import static com.blitzapp.animatedsplash.animation.Splash.shouldHide;

public class ObjectAnimation {

    public String animationType;
    private int animationDuration;
    private boolean setFillAfter;
    // Translate animation
    private float fromXDelta;
    private float toXDelta;
    private float fromYDelta;
    private float toYDelta;
    public String next;
    public String hideNext;
    public int priority;
    public int hidePriority;

    public AnimatedObject object, nextObject;
    public AnimatedText textobject, textNextObject;

    public boolean isLastObject;
    public boolean isLoop;

    // for rotate animation
    private float startFromValue;
    private float endToValue;
    private boolean isText = false;
    private View animationView;

    private Boolean executed = false;

    public ObjectAnimation() {
    }

    // For translate animation
    public ObjectAnimation(AnimatedObject object, String animationType, int animationDuration, float fromXDelta, float toXDelta, float fromYDelta, float toYDelta, boolean loop) {
        this.animationType = animationType;
        this.animationDuration = animationDuration;
        this.setFillAfter = true;
        this.fromXDelta = fromXDelta;
        this.toXDelta = toXDelta;
        this.fromYDelta = fromYDelta;
        this.toYDelta = toYDelta;
        this.isLoop = loop;

        this.priority = priority;
        this.object = object;
        this.isText = false;
        Log.d(TAG, "AnimateObject: priority " + this.priority);
    }

    public ObjectAnimation(AnimatedObject object, String animationType, int animationDuration, float fromXDelta, float toXDelta, float fromYDelta, float toYDelta) {
        this.animationType = animationType;
        this.animationDuration = animationDuration;
        this.setFillAfter = true;
        this.fromXDelta = fromXDelta;
        this.toXDelta = toXDelta;
        this.fromYDelta = fromYDelta;
        this.toYDelta = toYDelta;
        this.priority = priority;
        this.object = object;
        this.isText = false;

    }

    // For rotate animation
    public ObjectAnimation(AnimatedObject object, String animationType, int animationDuration, float fromValue, float toValue, boolean loop) {
        this.animationType = animationType;
        this.animationDuration = animationDuration;
        this.setFillAfter = true;
        this.isLoop = loop;
        this.startFromValue = fromValue;
        this.endToValue = toValue;
        this.priority = priority;
        this.object = object;
        this.isText = false;

        Log.d(TAG, "AnimateObject: priority " + this.priority);
    }

    // For rotate animation
    public ObjectAnimation(AnimatedObject object, String animationType, int animationDuration, float fromValue, float toValue) {
        this.animationType = animationType;
        this.animationDuration = animationDuration;
        this.setFillAfter = true;
        this.startFromValue = fromValue;
        this.endToValue = toValue;
        this.priority = priority;
        this.object = object;
        this.isText = false;

    }


    public AnimatedObject getNextObject() {
        return nextObject;
    }

    public String getAnimationType() {
        return animationType;
    }

    public AnimatedObject getObject() {
        return object;
    }

    public AnimatedText getTextObject() {
        return textobject;
    }

    public int getAnimationDuration() {
        return animationDuration;
    }

    public boolean isSetFillAfter() {
        return setFillAfter;
    }

    public float getFromXDelta() {
        return fromXDelta;
    }

    public float getToXDelta() {
        return toXDelta;
    }

    public float getFromYDelta() {
        return fromYDelta;
    }

    public float getToYDelta() {
        return toYDelta;
    }

    public boolean isLastObject() {
        return isLastObject;
    }

    public int getPriority() {
        return priority;
    }

    public boolean isTextType() {
        return isText;
    }

    public float getRotateStartDegree() {
        return startFromValue;
    }

    public float getRotateEndDegree() {
        return endToValue;
    }

    public Boolean getExecuted() {
        return executed;
    }

    public void setExecuted(Boolean executed) {
        this.executed = executed;
    }

    private void checkGroup(ObjectAnimation object) {
        if (object.next != null) {
            Log.d("ANIMATED_SPLASH", object.priority + "==" + animationsList.get(object.next).priority);
            if (object.priority == animationsList.get(object.next).priority) {
                runSpecificAnimation(animationsList.get(object.next));
            }
        }
    }

    private void checkHideGroup(ObjectAnimation object) {
        if (object.hideNext != null) {
            if (object.hidePriority == hideAnimationsList.get(object.hideNext).hidePriority) {
                runHideAnimation(hideAnimationsList.get(object.hideNext));
            }
        }
    }

    private void runNextAnimation(ObjectAnimation object) {
        if (object.next != null) {
            runSpecificAnimation(animationsList.get(object.next));
        } else {
            allExecuted = true;
            if (!hideAnimationsList.isEmpty()) {
                runHideAnimation(hideAnimationsList.get(hideAnimationListHead));
            }else{
                allHideExecuted = true;
                shouldHide = true;
                animationhide();
            }

        }
    }

    private void runNextHideAnimation(ObjectAnimation object) {
        if (object.hideNext != null) {

            runHideAnimation(hideAnimationsList.get(object.hideNext));
        } else {
            allHideExecuted = true;
            shouldHide = true;
            animationhide();


        }
    }

    public void slideAnimation(ObjectAnimation object) {

        View view = object.object.getImageView();

        TranslateAnimation slideImage = new TranslateAnimation(fromXDelta, toXDelta, fromYDelta, toYDelta);
        slideImage.setDuration(animationDuration);
        slideImage.setFillAfter(setFillAfter);
        view.setVisibility(View.VISIBLE);
        view.startAnimation(slideImage);

        checkGroup(object);

        if (object.isLoop) {
            slideImage.setRepeatCount((int) Double.POSITIVE_INFINITY);
            slideImage.setRepeatMode(ValueAnimator.REVERSE);

        }

        slideImage.setAnimationListener(new Animation.AnimationListener() {
            @Override
            public void onAnimationStart(Animation animation) {

            }

            @Override
            public void onAnimationEnd(Animation animation) {
                runNextAnimation(object);
            }


            @Override
            public void onAnimationRepeat(Animation animation) {
                runNextAnimation(object);


            }
        });

    }

    public void rotateAnimation(ObjectAnimation object) {
        View view = object.object.getImageView();

        ObjectAnimator rotation = ObjectAnimator.ofFloat(view, View.ROTATION, startFromValue, endToValue);
        rotation.setInterpolator(new LinearInterpolator());
        rotation.setDuration(animationDuration);
        view.setVisibility(View.VISIBLE);
        checkGroup(object);

        if (object.isLoop) {
            rotation.setRepeatCount(ValueAnimator.INFINITE);
            rotation.setRepeatMode(ValueAnimator.REVERSE);
            rotation.setInterpolator(new LinearInterpolator());
        }
        rotation.start();

        rotation.addListener(new AnimatorListenerAdapter() {

            @Override
            public void onAnimationRepeat(Animator animation) {
                super.onAnimationRepeat(animation);
                runNextAnimation(object);
            }

            @Override
            public void onAnimationEnd(Animator animation) {
                super.onAnimationEnd(animation);
                runNextAnimation(object);
            }
        });
    }

    public void fadeAnimation(ObjectAnimation object) {

        View view = object.object.getImageView();
        view.setAlpha(1);
        ObjectAnimator fadeIn = ObjectAnimator.ofFloat(view, "alpha", startFromValue, endToValue);
        fadeIn.setInterpolator(new LinearInterpolator()); //add this
        fadeIn.setDuration(animationDuration);
        view.setVisibility(View.VISIBLE);
        checkGroup(object);

        if (object.isLoop) {
            fadeIn.setRepeatCount(ValueAnimator.INFINITE);
            fadeIn.setRepeatMode(ValueAnimator.REVERSE);
        }
        fadeIn.start();

        fadeIn.addListener(new AnimatorListenerAdapter() {
            @Override
            public void onAnimationStart(Animator animation) {

            }

            @Override
            public void onAnimationEnd(Animator animation) {
                runNextAnimation(object);
            }


            @Override
            public void onAnimationRepeat(Animator animation) {
                Log.d("ANIMATED_SPLASH", object.toString());
                runNextAnimation(object);

            }
        });

    }

    public void scaleAnimation(ObjectAnimation object) {
        View view = object.object.getImageView();
        ObjectAnimator scale = ObjectAnimator.ofPropertyValuesHolder(view,
                PropertyValuesHolder.ofFloat("scaleX", fromXDelta, toXDelta),
                PropertyValuesHolder.ofFloat("scaleY", fromYDelta, toYDelta));
        scale.setDuration(animationDuration);
        view.setVisibility(View.VISIBLE);
        checkGroup(object);

        if (object.isLoop) {
            scale.setRepeatCount(ValueAnimator.INFINITE);
            scale.setRepeatMode(ValueAnimator.REVERSE);
        }
        scale.start();

        scale.addListener(new AnimatorListenerAdapter() {
            @Override
            public void onAnimationStart(Animator animation) {

            }

            @Override
            public void onAnimationEnd(Animator animation) {
                runNextAnimation(object);

            }

            @Override
            public void onAnimationRepeat(Animator animation) {
                runNextAnimation(object);

            }
        });
    }

    public void slideHideAnimation(ObjectAnimation object) {

        View view = object.object.getImageView();

        TranslateAnimation slideImage = new TranslateAnimation(fromXDelta, toXDelta, fromYDelta, toYDelta);
        slideImage.setDuration(animationDuration);
        slideImage.setFillAfter(setFillAfter);
        view.setVisibility(View.VISIBLE);
        view.startAnimation(slideImage);

        checkHideGroup(object);

        slideImage.setAnimationListener(new Animation.AnimationListener() {
            @Override
            public void onAnimationStart(Animation animation) {

            }

            @Override
            public void onAnimationEnd(Animation animation) {
                runNextHideAnimation(object);
            }

            @Override
            public void onAnimationRepeat(Animation animation) {

            }

        });

    }

    public void rotateHideAnimation(ObjectAnimation object) {
        View view = object.object.getImageView();

        ObjectAnimator rotation = ObjectAnimator.ofFloat(view, View.ROTATION, startFromValue, endToValue);
        rotation.setInterpolator(new LinearInterpolator());
        rotation.setDuration(animationDuration);
        view.setVisibility(View.VISIBLE);
        checkHideGroup(object);
        rotation.start();

        rotation.addListener(new AnimatorListenerAdapter() {
            @Override
            public void onAnimationRepeat(Animator animation) {
                super.onAnimationRepeat(animation);
            }

            @Override
            public void onAnimationEnd(Animator animation) {
                super.onAnimationEnd(animation);
                runNextHideAnimation(object);

            }
        });
    }

    public void fadeHideAnimation(ObjectAnimation object) {

        View view = object.object.getImageView();
        view.setAlpha(1);
        ObjectAnimator fadeIn = ObjectAnimator.ofFloat(view, "alpha", startFromValue, endToValue);
        fadeIn.setInterpolator(new LinearInterpolator()); //add this
        fadeIn.setDuration(animationDuration);
        view.setVisibility(View.VISIBLE);
        checkHideGroup(object);

        fadeIn.start();

        fadeIn.addListener(new AnimatorListenerAdapter() {
            @Override
            public void onAnimationStart(Animator animation) {
            }

            @Override
            public void onAnimationEnd(Animator animation) {
                runNextHideAnimation(object);

            }


            @Override
            public void onAnimationRepeat(Animator animation) {
            }
        });
    }

    public void scaleHideAnimation(ObjectAnimation object) {
        View view = object.object.getImageView();
        ObjectAnimator scale = ObjectAnimator.ofPropertyValuesHolder(view,
                PropertyValuesHolder.ofFloat("scaleX", fromXDelta, toXDelta),
                PropertyValuesHolder.ofFloat("scaleY", fromYDelta, toYDelta));
        scale.setDuration(animationDuration);
        view.setVisibility(View.VISIBLE);
        checkHideGroup(object);

        scale.start();

        scale.addListener(new AnimatorListenerAdapter() {
            @Override
            public void onAnimationStart(Animator animation) {
            }

            @Override
            public void onAnimationEnd(Animator animation) {
                runNextHideAnimation(object);
            }

            @Override
            public void onAnimationRepeat(Animator animation) {
            }
        });
    }

    public String getNext() {
        return next;
    }

    public void setNext(String next) {
        this.next = next;
    }

    public String getHideNext() {
        return hideNext;
    }

    public void setHideNext(String hideNext) {
        this.hideNext = hideNext;
    }

    public void setPriority(int priority) {
        this.priority = priority;
    }

    public int getHidePriority() {
        return hidePriority;
    }

    public void setHidePriority(int hidePriority) {
        this.hidePriority = hidePriority;
    }
}

File: ios/Animation/ObjectAnimation.m
//
//  ObjectAnimation.m
//  RNAnimatedSplash
//
//  Created by Ethan Gray on 8/24/22.
//

#import "ObjectAnimation.h"
#import "AnimatedObject.h"
#import <UIKit/UIKit.h>
#import <CoreGraphics/CoreGraphics.h>
#import "Splash.h"
#import "AnimationsList.h"
#import "HideAnimationsList.h"
@class Splash;
@class AnimatedObject;
@class AnimationList;
@implementation ObjectAnimation


//Written By Ayesh

-(id)initimage:(AnimatedObject *)object animationtype:(int) animationtype animationDuration:(float)animationDuration fromVal:(CGFloat)fromVal toVal:(CGFloat)toVal{
    _imageview = object;
    _animationType = animationtype;
    _animationDuration = animationDuration/1000;
    _toValue = toVal;
    _fromValue = fromVal;
    return self;
}

-(id)initimage:(AnimatedObject *)object animationtype:(int) animationtype animationDuration:(float)animationDuration fromX:(CGFloat)fromX toX:(CGFloat)toX fromY:(CGFloat)fromY toY:(CGFloat)toY{
    _imageview = object;
    _animationType = animationtype;
    _animationDuration = animationDuration/1000;
    _fromXDelta = fromX;
    _toXDelta = toX;
    _fromYDelta = fromY;
    _toYDelta = toY;
    return self;
}

-(id)initimage:(AnimatedObject *)object animationtype:(int) animationtype animationDuration:(float)animationDuration rotateDegree:(float)rotateDegree{
    _imageview = object;
    _animationType = animationtype;
    _animationDuration = animationDuration/1000;
    _fromValue = 0;
    _toValue = rotateDegree;
    return self;
}

-(id)initimage:(AnimatedObject *)object animationtype:(int) animationtype animationDuration:(float)animationDuration scaleX:(CGFloat)scaleX scaleY:(CGFloat)scaleY{
    _imageview = object;
    _animationType = animationtype;
    _animationDuration = animationDuration/1000;
    _fromValue = 0;
    _scaleX = scaleX;
    _scaleY = scaleY;
    return self;
}


//With Loop
-(id)initimage:(AnimatedObject *)object animationtype:(int) animationtype animationDuration:(float)animationDuration fromVal:(CGFloat)fromVal toVal:(CGFloat)toVal loop:(Boolean)loop{
    _imageview = object;
    _animationType = animationtype;
    _animationDuration = animationDuration/1000;
    _toValue = toVal;
    _fromValue = fromVal;
    _isLoop = loop;
    return self;
}

-(id)initimage:(AnimatedObject *)object animationtype:(int) animationtype animationDuration:(float)animationDuration fromX:(CGFloat)fromX toX:(CGFloat)toX fromY:(CGFloat)fromY toY:(CGFloat)toY loop:(Boolean)loop{
    _imageview = object;
    _animationType = animationtype;
    _animationDuration = animationDuration/1000;
    _fromXDelta = fromX;
    _toXDelta = toX;
    _fromYDelta = fromY;
    _toYDelta = toY;
    _isLoop = loop;
    return self;
}

-(id)initimage:(AnimatedObject *)object animationtype:(int) animationtype animationDuration:(float)animationDuration rotateDegree:(float)rotateDegree loop:(Boolean)loop{
    _imageview = object;
    _animationType = animationtype;
    _animationDuration = animationDuration/1000;
    _fromValue = 0;
    _toValue = rotateDegree;
    _isLoop = loop;
    return self;
}

-(id)initimage:(AnimatedObject *)object animationtype:(int) animationtype animationDuration:(float)animationDuration scaleX:(CGFloat)scaleX scaleY:(CGFloat)scaleY loop:(Boolean)loop{
    _imageview = object;
    _animationType = animationtype;
    _animationDuration = animationDuration/1000;
    _fromValue = 0;
    _scaleX = scaleX;
    _scaleY = scaleY;
    _isLoop = loop;
    return self;
}

-(AnimatedObject *)getObject{
    return _imageview;
}
-(int)getAnimationtype{
    return _animationType;
}
-(int)getPriority{
    return _priority;
}
-(bool)getisLoop{
    return _isLoop;
}

-(void) runNextAnimation:(ObjectAnimation *)object {
    if (object.next != NULL) {
        if([AnimationsList getItem:object.next].executed == false){
            [splashClassInstance runSpecificAnimation: [AnimationsList getItem:object.next]];
        }
    }else{
        allExecuted = true;
        if([HideAnimationsList getLength]>0){
            ObjectAnimation *obj = [HideAnimationsList getItem:[HideAnimationsList getHead]];
            [splashClassInstance runSpecificHideAnimation:obj];
        }else{
            if(splashClassInstance){
                [splashClassInstance dismissSplashDialog];
            }
        }
        
        
        
    }
}

-(void) runNextHideAnimation:(ObjectAnimation *)object {
    if (object.hideNext != NULL) {
        if([HideAnimationsList getItem:object.hideNext].hideExecuted == false){
            [splashClassInstance runSpecificHideAnimation: [HideAnimationsList getItem:object.hideNext]];
        }
    }else{
        
        shouldHide = true;
        if([HideAnimationsList getLength] > 0){
            if(splashClassInstance){
                [splashClassInstance dismissSplashDialog];
            }
        }else{
            return;
        }
        
        
    }
}

-(void)slideAnimation:(ObjectAnimation *)imageobject{
    
    NSLog(@"priority %d",_priority);
    dispatch_async(dispatch_get_main_queue(), ^{
        UIImageView *imagev = imageobject.imageview.getImageView;
        //        imagev.transform = CGAffineTransformTranslate(imagev.transform,_fromXDelta, _fromYDelta);
        imagev.hidden = NO;
        [UIView animateWithDuration:imageobject.animationDuration delay:0.0 options:(imageobject.isLoop?UIViewAnimationOptionRepeat| UIViewKeyframeAnimationOptionAutoreverse:nil ) animations:^{
            imagev.transform = CGAffineTransformTranslate(imagev.transform,_toXDelta, _toYDelta);
        } completion:^(BOOL finished){
            [imageobject setExecuted:true];
            [self runNextAnimation:imageobject];
        }
        ];
        if ([AnimationsList getItem:imageobject.next].priority == imageobject.priority){
            [imageobject setExecuted:true];
            [self runNextAnimation:imageobject];
        }
        else if(imageobject.isLoop){
            dispatch_time_t delay = dispatch_time(DISPATCH_TIME_NOW, NSEC_PER_SEC * imageobject.animationDuration);
            dispatch_after(delay, dispatch_get_main_queue(), ^(void){
                // do work in the UI thread here
                [self runNextAnimation:imageobject];
            });
            
        }
    });
}
-(void)scaleAnimation:(ObjectAnimation *)imageobject{
    NSLog(@"priority %d",_priority);
    dispatch_async(dispatch_get_main_queue(), ^{
        UIImageView *imagev = imageobject.imageview.getImageView;
        imagev.hidden = NO;
        //     imagev.transform = CGAffineTransformScale(imagev.transform,_fromXDelta, _fromYDelta);
        [UIView animateWithDuration:imageobject.animationDuration delay:0.0 options:(imageobject.isLoop?UIViewAnimationOptionRepeat| UIViewKeyframeAnimationOptionAutoreverse:nil ) animations:^{
            imagev.transform = CGAffineTransformScale(imagev.transform,_scaleX, _scaleY);
        } completion:^(BOOL finished){
            [imageobject setExecuted:true];
            [self runNextAnimation:imageobject];
        }];
        if ([AnimationsList getItem:imageobject.next].priority == imageobject.priority){
            [imageobject setExecuted:true];
            [self runNextAnimation:imageobject];
        }
        else if(imageobject.isLoop){
            dispatch_time_t delay = dispatch_time(DISPATCH_TIME_NOW, NSEC_PER_SEC * imageobject.animationDuration);
            dispatch_after(delay, dispatch_get_main_queue(), ^(void){
                // do work in the UI thread here
                [self runNextAnimation:imageobject];
            });
            
        }
    });
}
-(void)rotateAnimation:(ObjectAnimation *)imageobject{
    dispatch_async(dispatch_get_main_queue(), ^{
        
        UIImageView *imagev = imageobject.imageview.getImageView;
        
        imagev.hidden = NO;
        imagev.transform = CGAffineTransformRotate(imagev.transform,_fromValue * M_PI/360.0);
        [UIView animateWithDuration:imageobject.animationDuration delay:0.0 options:(imageobject.isLoop?UIViewAnimationOptionRepeat| UIViewKeyframeAnimationOptionAutoreverse:nil ) animations:^{
            imagev.transform = CGAffineTransformRotate(imagev.transform,_toValue * M_PI/360.0);
        } completion:^(BOOL finished){
            [imageobject setExecuted:true];
            [self runNextAnimation:imageobject];
        }];
        if ([AnimationsList getItem:imageobject.next].priority == imageobject.priority){
            [imageobject setExecuted:true];
            [self runNextAnimation:imageobject];
        }
        else if(imageobject.isLoop){
            dispatch_time_t delay = dispatch_time(DISPATCH_TIME_NOW, NSEC_PER_SEC * imageobject.animationDuration);
            dispatch_after(delay, dispatch_get_main_queue(), ^(void){
                // do work in the UI thread here
                [self runNextAnimation:imageobject];
            });
            
        }
    });
}
-(void)fadeAnimation:(ObjectAnimation *)imageobject{
    dispatch_async(dispatch_get_main_queue(), ^{
        UIImageView *imagev = imageobject.imageview.getImageView;
        imagev.alpha = self->_fromValue;
        imagev.hidden = NO;
        [UIView animateWithDuration:imageobject.animationDuration delay:0.0 options:(imageobject.isLoop?UIViewAnimationOptionRepeat| UIViewKeyframeAnimationOptionAutoreverse:nil) animations:^{
            imagev.alpha = self->_toValue;
        } completion:^(BOOL finished){
            [imageobject setExecuted:true];
            [self runNextAnimation:imageobject];
        }
        ];
        if ([AnimationsList getItem:imageobject.next].priority == imageobject.priority){
            [imageobject setExecuted:true];
            [self runNextAnimation:imageobject];
        }
        else if(imageobject.isLoop){
            dispatch_time_t delay = dispatch_time(DISPATCH_TIME_NOW, NSEC_PER_SEC * imageobject.animationDuration);
            dispatch_after(delay, dispatch_get_main_queue(), ^(void){
                // do work in the UI thread here
                [self runNextAnimation:imageobject];
            });
            
        }
        
    });
}


-(void)slideHideAnimation:(ObjectAnimation *)imageobject{
    
    dispatch_async(dispatch_get_main_queue(), ^{
        UIImageView *imagev = imageobject.imageview.getImageView;
        imagev.transform = CGAffineTransformTranslate(imagev.transform,_fromXDelta, _fromYDelta);
        imagev.hidden = NO;
        [UIView animateWithDuration:imageobject.animationDuration delay:0.0 options:nil animations:^{
            imagev.transform = CGAffineTransformTranslate(imagev.transform,_toXDelta, _toYDelta);
        } completion:^(BOOL finished){
            [imageobject setHideExecuted:true];
            [self runNextHideAnimation:imageobject];
        }
        ];
        if ([HideAnimationsList getItem:imageobject.hideNext].hidePriority == imageobject.hidePriority){
            [imageobject setHideExecuted:true];
            [self runNextHideAnimation:imageobject];
        }
    });
}
-(void)scaleHideAnimation:(ObjectAnimation *)imageobject{
    dispatch_async(dispatch_get_main_queue(), ^{
        UIImageView *imagev = imageobject.imageview.getImageView;
        imagev.hidden = NO;
        //     imagev.transform = CGAffineTransformScale(imagev.transform,_fromXDelta, _fromYDelta);
        [UIView animateWithDuration:imageobject.animationDuration delay:0.0 options:nil animations:^{
            imagev.transform = CGAffineTransformScale(imagev.transform,_scaleX, _scaleY);
        } completion:^(BOOL finished){
            [imageobject setHideExecuted:true];
            [self runNextHideAnimation:imageobject];
        }];
        if ([HideAnimationsList getItem:imageobject.hideNext].hidePriority == imageobject.hidePriority){
            [imageobject setHideExecuted:true];
            [self runNextHideAnimation:imageobject];
        }
        
    });
}
-(void)rotateHideAnimation:(ObjectAnimation *)imageobject{
    dispatch_async(dispatch_get_main_queue(), ^{
        
        UIImageView *imagev = imageobject.imageview.getImageView;
        
        imagev.hidden = NO;
        imagev.transform = CGAffineTransformRotate(imagev.transform,_fromValue * M_PI/360.0);
        [UIView animateWithDuration:imageobject.animationDuration delay:0.0 options:nil animations:^{
            imagev.transform = CGAffineTransformRotate(imagev.transform,_toValue * M_PI/360.0);
        } completion:^(BOOL finished){
            [imageobject setHideExecuted:true];
            [self runNextHideAnimation:imageobject];
        }];
        if ([HideAnimationsList getItem:imageobject.hideNext].hidePriority == imageobject.hidePriority){
            [imageobject setHideExecuted:true];
            [self runNextHideAnimation:imageobject];
        }
    });
}
-(void)fadeHideAnimation:(ObjectAnimation *)imageobject{
    dispatch_async(dispatch_get_main_queue(), ^{
        UIImageView *imagev = imageobject.imageview.getImageView;
        imagev.alpha = self->_fromValue;
        imagev.hidden = NO;
        [UIView animateWithDuration:imageobject.animationDuration delay:0.0 options:nil animations:^{
            imagev.alpha = self->_toValue;
        } completion:^(BOOL finished){
            [imageobject setHideExecuted:true];
            [self runNextHideAnimation:imageobject];
        }
        ];
        if ([HideAnimationsList getItem:imageobject.hideNext].hidePriority == imageobject.hidePriority){
            [imageobject setHideExecuted:true];
            [self runNextHideAnimation:imageobject];
        }
    });
}


@end
